test/unit/Elsa.Activities.UnitTests/Branching/IfTests.cs

```diff
@@ -0,0 +1,107 @@
+using Elsa.Extensions;
+using Elsa.Testing.Shared;
+using Elsa.Workflows;
+
+namespace Elsa.Activities.UnitTests.Branching;
+
+public class IfTests
+{
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async Task Should_Set_Result_To_Condition_Value_Regardless_Of_Branch_Presence(bool conditionValue)
+    {
+        // Arrange - Test with no branches to verify result is independent of branch activities
+        var ifActivity = new If(() => conditionValue);
+
+        // Act
+        var context = await ExecuteAsync(ifActivity);
+
+        // Assert
+        var resultValue = (bool)context.GetActivityOutput(() => ifActivity.Result)!;
+        Assert.Equal(conditionValue, resultValue);
+    }
+
+    [Theory]
+    [InlineData(true, true, false)] // condition true, has then branch, no else branch
+    [InlineData(false, false, true)] // condition false, no then branch, has else branch
```

Comment:
Tests only 2 out of 4 possible branch configurations:
Missing: condition=true, hasThen=false, hasElse=true
Missing: condition=false, hasThen=true, hasElse=false

```suggestion
    [InlineData(true, true, false)] // condition true, has then branch, no else branch
    [InlineData(false, false, true)] // condition false, no then branch, has else branch
    [InlineData(true, false, true)] // condition true, no then branch, has else branch
    [InlineData(false, true, false)] // condition false, has then branch, no else branch
```